content/departments/product-engineering/engineering/cloud/delivery/managed/suspend_process.md

```diff
@@ -0,0 +1,104 @@
+
+Managed instances configuration is tracked in [`deploy-sourcegraph-managed-instances`](https://github.com/sourcegraph/deploy-sourcegraph-managed).
+
+For basic operations like accessing an instance for these steps, see [managed instances operations](operations.md).
+
+- [General setup](#general-setup)
+- [Sourcegraph suspend](#sourcegraph-suspend)
+  - [0) Sourcegraph suspend setup](#0-sourcegraph-suspend-setup)
+  - [1) Mark the database as ready-only](#1-mark-the-database-as-read-only)
+  - [2) Create a snapshot of the current deployment](#2-create-a-snapshot-of-the-current-deployment)
+  - [3) Remove some resources](#3-remove-some-resources)
+  - [4) Wrapping up](#4-wrapping-up)
+
+## General setup
+
+Managed instances configuration is tracked in [`deploy-sourcegraph-managed`](https://github.com/sourcegraph/deploy-sourcegraph-managed) - make sure you have the latest revision of this repository checked out. For basic operations like accessing an instance for these steps, see [managed instances operations](operations.md).
+
+First, ensure you have the prerequisites installed and up-to-date:
+
+- [Terraform CLI](https://www.terraform.io/)
+- [Deno CLI](https://deno.land/) (used for scripting upgrades)
+- [Comby CLI](https://comby.dev/) (used for rewriting configuration)
+- [jq](https://stedolan.github.io/jq/)
+
+Many of the following commands in this guide, as well as the commands [operations](./operations.md), use environment variables. Export the appropriate values for the upgrade so you don't lose track:
+
+```sh
+# name of customer deployment (should match folder)
+export CUSTOMER=<customer_or_instance_name>
+export CURRENT_DEPLOYMENT=$(gcloud compute instances list --project=sourcegraph-managed-${CUSTOMER} | grep -v "executors" | awk 'NR>1 { if ($1 ~ "-red-") print "red"; else print "black"; }')
+```
+
+Make sure your copy of the [`deploy-sourcegraph-managed`](https://github.com/sourcegraph/deploy-sourcegraph-managed) repository is up to date:
+
+```sh
+git checkout main
+git pull origin main
+```
+
+## Sourcegraph suspend
+
+### 0) Sourcegraph suspend setup
+
+**Follow the [general setup](#general-setup) guide.** Then, set up the appropriate version variables:
+
+Make sure to use the same shell for all the commands in this guide unless otherwise stated.
+
+Now start a branch for your upgrade:
+
+```sh
+git checkout -b $CUSTOMER/suspend-instance
+# all the below steps are documented assuming you are in the customer deployment directory
+cd $CUSTOMER
+```
+
+### 1) Mark the database as ready-only
+
+```sh
+../util/set-db-readonly.sh $CURRENT_DEPLOYMENT true
+```
+
+### 2) Create a snapshot of the current deployment
+
+```sh
+../util/create-snapshot.ts $CURRENT_DEPLOYMENT
+git add . && git commit -m "$CUSTOMER: snapshot deployment"
+```
+
+This can take anywhere from a minute to several minutes, depending on how large the disk is.
+
+Note the number of snapshots in `terraform.tfvars`, and prune snapshots >5 upgrades old where appropriate.
```

Comment:
In this situation, we may want to go ahead and remove all snapshots except this last one. I suppose it can't hurt to keep at least one more though. Thoughts?

Reply:
good point. we don't need all these past snapshots anyway.